vm/env.go

```diff
@@ -422,3 +422,32 @@ func (e *Env) Execute(src string) (interface{}, error) {
 func (e *Env) Run(stmts []ast.Stmt) (interface{}, error) {
 	return Run(stmts, e)
 }
+
+// Copy the state of the virtual machine environment
+func (e *Env) Copy() *Env {
+	b := false
+	copy := Env {
+		env:       make(map[string]reflect.Value),
+		typ:       make(map[string]reflect.Type),
+		parent:    e.parent,
+		interrupt: &b,
+		name:      e.name,
+		external:  e.external,
+	}
+	for name, value := range e.env {
+		copy.env[name] = value
+	}
+	for name, typ := range e.typ {
+		copy.typ[name] = typ
+	}
```

Comment:
Should there be a lock when coping the maps?

Reply:
Added lock on copy.